react/README.md

```diff
@@ -73,15 +87,90 @@ For integration tests, we mock the Apollo client with a `MockedProvider` compone
-* No index.ts files, besides the entry file for the app
-* Ideally only one component per file
-* Files and folders which export a component/view are written UpperCamelCase, with the same name as the actual exported component/view
-* Other files (like types.ts, helpers.ts etc) and folders (like providers, utils etc) are written in lowerCamelCase
-* Config constants should be UPPERCASE_SNAKES
-* GraphQL queries, mutation and subscription  **string**  have the format UPPERCASE_SNAKES_<QUERY|MUTATION|SUBSCRIPTION>
+-   Views of react-router paths go into the views folder
+-   Each view can have it's own folder - which in return can have a local components folder
+-   Following an "As local as possible, as global as needed" approach: components get only moved into a more global/outer folder if they are used on that level
+-   No index.ts files, besides the entry file for the app
+-   Ideally only one component per file
+-   Files and folders which export a component/view are written UpperCamelCase, with the same name as the actual exported component/view
+-   Other files (like types.ts, helpers.ts etc) and folders (like providers, utils etc) are written in lowerCamelCase
+-   Config constants should be UPPERCASE_SNAKES
+-   GraphQL queries, mutation and subscription **string** have the format UPPERCASE*SNAKES*<QUERY|MUTATION|SUBSCRIPTION>
+
+## Folders and files structures
+
+The following rules and naming conversions can be used to name files:
+
+```bash
+|---- <NameOfComponent>Container.ts # GraphQL string definitions, Business Logic, Data transformation
+|
+|---- <NameOfComponent>.ts # only UI parts **views**
+|---- <NameOfView>ViewContainer.tsx # main entry file the ReactRouter is referring to for a whole page including (maybe) GraphQL string definitions, extraction of URL params, ReactContext definitions (if needed), Business Logic, Data transformation
+|---- <NameOfView>Query.ts # if GraphQL definitions are too long and should be singled out
+|---- <NameOfView>Utils.ts # definitions of custom Hooks
+|---- /components # subcomponents only relevant to this page
+|--------<NameOfComponent>Container.ts # GraphQL string definitions, Business Logic, Data transformation
+|--------<NameOfComponent>.ts # **only** UI parts
+```
+
+The folder structure is as follows:
+
+```bash
+
+/react
+├── /src
+│   ├── assets
+│   ├── components
+│   │   ├── <NameOfComponent>
+│   │   |    ├── <NameOfComponent>Container.tsx
+│   |   |    └── <NameOfComponent>.stoties.tsx #storybook component definition
+│   |   └── Layout.tsx # main layout
+│   ├── mocks
+│   ├── providers #context providers
+│   ├── types #typescript definitions
+│   ├── views
+│   │   └── <NameOfComponent>
+│   │       ├──components
+│   |       └── <NameOfSubComponent>.tsx
+│   └── utils
+│           ├── bese-types.ts
+│           ├── helpers.ts
+│           ├── queries.ts
+│           ├── test-utils.ts
+│           └── hooks.ts
+├── node_modules
```

Comment:
Not sure if I'd list the entire content here, it quickly becomes out-of-sync. If people want to have an overview, they should `ls` or see their file-explorer. It makes sense for the `src/` part though.

Reply:
@pylipp I thought that since most of these files are used for configuration and are unlikely to change, it could be helpful to show them in the docs, but I totally agree with you that they can be easily verified via `ls` or file-explorer.